.github/workflows/tests_and_formatting.yml

```diff
@@ -74,6 +74,11 @@ jobs:
       - name: Check if test succeeded
         if: steps.test.outcome != 'success'
         run: exit 1
+      - name: Check out
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.head_ref }}
+          fetch-depth: 0
```

Comment:
Why have you added this? We already have another checkout step at line number 14.

Reply:
The "prettier-action" requires fetch-depth: 0 to work with the only_changed: True option enabled. Fetching with a depth of 0 is resource-intensive, which is why I am executing it only after all Rust checks have passed.

Reply:
Make fetch-depth 0 at the top. You are trying to make it fast I understand, but I hope you understand that your end result is making it slow.

Reply:
Got it. I am creating a separate PR for this fix.

Reply:
Also fetch-depth 0 should be used if you are trying to be smart and only run prettier on files that have modified, it does not make sense for us, we have hardly ~1K lines of code to check, no matter how slow prettier is, it will be faster to recheck all files instead of downloading the entire git history just so we can find what changed.

Reply:
Also use https://github.com/dprint/check, its written in Rust, will probably be faster to install (only a binary) and run.